time_interp/time_interp_external2.F90

```diff
@@ -1009,6 +1016,271 @@ subroutine time_interp_external_0d(index, time, data, verbose)
 
     end subroutine time_interp_external_0d
 
+
+    subroutine time_interp_external_bridge_2d(index1, index2, time, data_in, interp, verbose,horz_interp, mask_out, &
+               is_in, ie_in, js_in, je_in, window_id)
+
+      integer, intent(in) :: index1, index2
+      type(time_type), intent(in) :: time
+      real, dimension(:,:), intent(inout) :: data_in
```

Comment:
If this PR is going to be approved and merged into main soon, we can put in the mixed-precision updates.  Else, we've already changed time_interp to include mixed precision capabilities  and perhaps it would be easier to start off from the [time_interp_external.inc ](https://github.com/NOAA-GFDL/FMS/blob/mixedmode/time_interp/include/time_interp_external.inc) file in the mixedmode branch.  If possible, most of the mixed-precision updates (including data_override) will be merged in to the main branch by end of August and it would be simpler to wait until that merge before the changes here are included.  One thing to note is the file structures have been reorganized significantly with the mixed-precision updates and this PR will require more work to be compatible with main then.  What do you think?